### ISSUE TITLE
set country doesn't work without leading '+' when nationalMode is false

### ISSUE THREAD

**Issue by pduey** [jackocnr/intl-tel-input#473]
With the defunct autoFormat feature, there was a function _ensurePlus that would ensure the '+' isn't removed when nationalMode is false. That is kind of critical since, when nationalMode is false, we don't necessarily expect the user to select the country, we expect them to type in the country code and let the plugin set the country.

However, now, for example, if a user uses the delete key to erase the number, including the leading '+', then types in a valid number without the leading '+', the country is not set.

I get the reasons for removing autoFormat, but if nationalMode is false, we should still get "ensurePlus". Or perhaps the leading '+' could be optional (not my preference).

If you agree, I will do a pull request.

**Comment by jackocnr** [comment]
I'm sorry but I don't understand. Can you please provide steps to reproduce the issue? (And mention which initialisation options you're using)

**Comment by pduey** [comment]
Here's the basic config:
```
    allowExtensions: false
    autoFormat: true
    autoHideDialCode: true
    autoPlaceholder: true
    customPlaceholder: null
    defaultCountry:  ""
    geoIpLookup: null
    nationalMode: false

```
I set onlyCountries and preferredCountries in the markup.

To reproduce, add the intl-tel-input somewhere, put the cursor in the field, and use backspace/delete or select+cut to clear the input field, including the leading '+' sign (which is included initially  with nationalMode: false).

In an older version, before autoFormat was removed, you could not remove the leading '+' with nationalMode: false. There was a function to put it back in, even if it was cleared. 

Here's where it was removed:
```
git diff d782186f8696fd8ba7791f6f65d74046713f9f91..cf393033cb8bebc26dbdeb9fa22b6ca552f9c9d5
```
The commit comment is 
```
    removed formatAsYouType, allowExtensions and getExtension
```
Maybe I meant formatAsYouType, not defaultFormat?

Anyway, I put the function back, and call it here:
```
        // initialize any key listeners
        _initKeyListeners: function() {
            var that = this;
            // update flag on keyup
            // (keep this listener separate otherwise the setTimeout breaks all the tests)
            this.telInput.on("keyup" + this.ns, function() {
->              that._ensurePlus();
                that._updateFlagFromNumber(that.telInput.val());
            });
        ...
```

It is working fine for me now.

**Comment by jackocnr** [comment]
Thanks for writing that up - now I understand. Yes I removed the code to ensure there's always a '+' in the input when I removed autoFormat/formatAsYouType, because I decided it was bad practice to manipulate the user's input in any way. Instead I think the right thing to do is to validate the number (e.g. on change, or on submit) and give the user a useful error message if there is a problem.

**Comment by pduey** [comment]
Is there a case where passing the number to libphonenumber without the leading ‘+’ makes sense? E.g., if nationalMode is true, there’s an input mask that includes the extraneous characters for human readability, but they are stripped and the leading + and country code added before passing to libphonenumber, yes? When nationalMode is false, can we think of the leading + as the input mask?

Here’s the actual problem I’m solving, which maybe I should have led with.

User enters a non NANP number (e.g., Canada or Dominican Republic which use 1 for the country code). The flag icon is correctly updated to that country. The user puts the cursor in the input, deletes the number, including the country code and leading +, enters a new US number, this time without the leading + and country code. The flag icon is not updated, because without the leading +, it does not detect a valid country.

We use nationalMode false because we capture numbers from any country. They cut-n-paste a lot, so we don’t want to force them to first select the country. However, most of the users are US based, so they don’t really think in those terms. We need to keep the training wheels on for them.

**Comment by jackocnr** [comment]
I'm still failing to see the problem. If they change the flag to a different +1 country (by clicking it, or typing a number), then they clear the input and type a US number - it doesn't matter if the wrong +1 country is selected - if you use `getNumber` you will still get exactly the same number.

Also as an aside: you say "We use nationalMode false because we capture numbers from any country" - I just wanted to be clear that you can capture numbers from any country using nationalMode set to true as well - it's just the user has to select the right country from the dropdown first. Then if you use `getNumber` you still get the full number including the international dial code.

**Comment by jackocnr** [comment]
Closing due to inactivity.

**Comment by pduey** [comment]
Sorry, I was away for a few weeks.

I understand both points. 

On the first, I know that all the methods work correctly regardless of which flag is displayed, but users don't necessarily get that, so they report it as a bug.

On the second, yes, it works with national mode true, we just don't want to require users to select the country from the dropdown, as long as they have the number formatted correctly with the leading '+'.

So, in short, what we really want is for the '+' to always be there, so the correct flag is always selected, but not require the user to manually select the country. I have been using my own fork with the _ensurePlus method added back in with no problems. If you strongly disagree with this feature, I'll keep using my own fork. Otherwise, I can submit a pull request.

**Comment by jackocnr** [comment]
As I mentioned, I think it's bad practice to tamper with the input value in any way, so no I don't want that feature back in the plugin.

If you're getting bug reports from users who type a national number (no leading '+') and expect the flag to automatically update... well that doesn't make any sense - it's just impossible... so I wouldn't worry too much about them. If you really want to do something, perhaps you could add your own validation when they submit the form (or on `blur`) to check the number starts with a '+', and if not, provide a descriptive error message.

Alternatively, you could try the `separateDialCode` option - that might be a nice compromise.

> On the second, yes, it works with national mode true, we just don't want to require users to select the country from the dropdown, as long as they have the number formatted correctly with the leading '+'.

Using `nationalMode` doesn't *require* users to select the country manually - they are welcome to type the full number, with the leading '+' and it will work just fine!
